### PR TITLE
Add listscripts RPC command handling

### PR DESCRIPTION
**Pull request by cjepson**
The listscripts RPC command has been added, allowing a user to dump
the redeem scripts contained by their wallet. The dumped redeem scripts
also include the 20-byte HASH160 script hash and the script hash
address.

### CODE REVIEW

File: waddrmgr/address.go [hunk 1]
```diff
@@ -498,14 +498,14 @@ func (a *scriptAddress) Script() ([]byte, error) {
 		return nil, managerError(ErrWatchingOnly, errWatchingOnly, nil)
 	}
 
-	a.manager.mtx.Lock()
-	defer a.manager.mtx.Unlock()
-
 	// Account manager must be unlocked to decrypt the script.
 	if a.manager.locked {
```
Comment (jrick) [inline waddrmgr/address.go]: this has to be protected by the mutex.

**Comment by cjepson** [comment]
Reverted the mutex locking code

**Comment by jcvernaleo** [comment]
Could you squash this one?

**Comment by alexlyp** [comment]
tACK, listscripts command appears to accurately return imported scripts

**Pull request merged**